### PR TITLE
Relationships

### PR DESCRIPTION
**Pull request by yegonkimutai**
In this PR, I did the following:
- [ ]  Create `Classroom` class and `related` it to students class.
- [ ] Create the `Book` class to store authors and title.
- [ ] Create `Rental` class to `link` the `Book` and `Student` class

### CODE REVIEW

File: persons.rb [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+  attr_reader :id, :rentals
+
   def initialize(id, age, parent_permission = 'true', name = 'Unknown')
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (tufoinnkuo10) [inline persons.rb:15]: - [ ] like I mentioned in the book class, please kindly add a method for adding rentals here as well. here, the person itself should be sent as a parameter to create the new rental, achieving the 'has-many' association`

File: student.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-
 class Student < Persons
-  def initialize(id, age, classroom, parent_permission = 'true', name = 'Unknown')
+  attr_accessor :classroom
+
+  def initialize(id, age, parent_permission = 'true', name = 'Unknown')
     super(id, age, parent_permission, name)
-    @classroom = classroom
+    @classroom = nil
   end
```
Comment (tufoinnkuo10) [inline student.rb:7]: - [ ] nice job so far. please kindly add a method that serves as the 'belongs-to' side ie a student belongs to a classroom. can see more explanation [here](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`)
eg

```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
```
Comment (tufoinnkuo10) [inline book.rb:8]: - [ ] great job initializing rentals. Please kindly ensure A method for adding rentals should be present.kindly ensur the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. you can refresh on this [lesson](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`)
eg

```ruby
def add_rental(person, date) 
    Rental.new(date, self, person)
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @yegonkimutai 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ relationships added

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @yegonkimutai, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title and summary. ✔️
- Has a class `Classroom` ✔️
- Has a class `Book` ✔️
- Has a class `Rental` ✔️
- `has-many / belongs-to` relationships between `Classroom` and `Student` ✔️
- `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**